src/trainings.py

```diff
@@ -1,36 +1,172 @@
-import matplotlib.pyplot as plt
+import time
 import torch
+import torch.nn as nn
+import torch.optim as optim
 import numpy as np
+import matplotlib.pyplot as plt
 from utils.Dataset import CustomImageDataset
-import time
 from utils.models import ConvNet_1, ConvNet_2, ResNet18
-from utils.utils import set_seed
-import torch.optim as optim
-import torch.nn as nn
-from utils.utils import transform_pipeline
+from utils.utils import set_seed, transform_pipeline
+
+# Function to train a model
+def train_model(model, model_name: str):
+    """
+    Trains the model on CIFAR-10, saves the best model, and plots losses.
+
+    :param model: Model to train (e.g., ResNet18).
+    :param model_name: Name used for saving model and results.
+    """
+
+    set_seed(42) # Set random seed for reproducibility
+    print("Training "+model_name)
+
+    # Initialize model, loss function, and optimizer
+    net = model()
```

Comment:
It will not work. If the model is passed as the argument, it will be initialized before, delete that line and use model instead of net everywhere.